includes/Modules/Ads/AMP_Tag.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Ads\AMP_Tag
+ *
+ * @package   Google\Site_Kit\Modules\Ads
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Ads;
+
+use Google\Site_Kit\Core\Modules\Tags\Module_AMP_Tag;
+use Google\Site_Kit\Core\Tags\Tag_With_Linker_Interface;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use Google\Site_Kit\Core\Tags\Tag_With_Linker_Trait;
+
+/**
+ * Class for AMP tag.
+ *
+ * @since  n.e.x.t
+ * @access private
+ * @ignore
+ */
+class AMP_Tag extends Module_AMP_Tag implements Tag_With_Linker_Interface {
+
+	use Method_Proxy_Trait, Tag_With_Linker_Trait;
+
+	/**
+	 * Sets the current home domain.
+	 *
+	 * @since 1.118.0
+	 *
+	 * @param string $domain Domain name.
+	 */
+	public function set_home_domain( $domain ) {
+		$this->home_domain = $domain;
+	}
+
+	/**
+	 * Registers tag hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		$render = $this->get_method_proxy_once( 'render' );
+
+		// Which actions are run depends on the version of the AMP Plugin
+		// (https://amp-wp.org/) available. Version >=1.3 exposes a
+		// new, `amp_print_analytics` action.
+		// For all AMP modes, AMP plugin version >=1.3.
+		add_action( 'amp_print_analytics', $render );
+		// For AMP Standard and Transitional, AMP plugin version <1.3.
+		add_action( 'wp_footer', $render, 20 );
+		// For AMP Reader, AMP plugin version <1.3.
+		add_action( 'amp_post_template_footer', $render, 20 );
+		// For Web Stories plugin.
+		add_action( 'web_stories_print_analytics', $render );
+
+		// Load amp-analytics component for AMP Reader.
+		$this->enqueue_amp_reader_component_script( 'amp-analytics', 'https://cdn.ampproject.org/v0/amp-analytics-0.1.js' );
+
+		$this->do_init_tag_action();
+	}
+
+	/**
+	 * Outputs gtag <amp-analytics> tag.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function render() {
+		$config = $this->get_tag_config();
+
+		$gtag_amp_opt = array(
+			'optoutElementId' => '__gaOptOutExtension',
+			'vars'            => array(
+				'gtag_id' => $this->tag_id,
+				'config'  => $config,
+			),
+		);
+
+		/**
+		 * Filters the gtag configuration options for the amp-analytics tag.
+		 *
+		 * You can use the {@see 'googlesitekit_gtag_opt'} filter to do the same for gtag in non-AMP.
+		 *
+		 * @since 1.24.0
+		 * @see https://developers.google.com/gtagjs/devguide/amp
+		 *
+		 * @param array $gtag_amp_opt gtag config options for AMP.
+		 */
+		$gtag_amp_opt_filtered = apply_filters( 'googlesitekit_amp_gtag_opt', $gtag_amp_opt );
+
+		// Ensure gtag_id is set to the correct value.
+		if ( ! is_array( $gtag_amp_opt_filtered ) ) {
+			$gtag_amp_opt_filtered = $gtag_amp_opt;
+		}
+
+		if ( ! isset( $gtag_amp_opt_filtered['vars'] ) || ! is_array( $gtag_amp_opt_filtered['vars'] ) ) {
+			$gtag_amp_opt_filtered['vars'] = $gtag_amp_opt['vars'];
+		}
```

Comment:
This filter is specific to Analytics so it shouldn't be present here. We've never had a filter specific for filtering the Ads gtag options so this would be consistent with that. This will be enhanced when we move it to Gtag.

Reply:
I've removed the filter and instead used `$gtag_amp_opt` directly 👍🏻